Update installation docs

knelli2 opened the pull request:
Small things I found while setting up a new environment

- Blaze version was out of date.
- If installing dependencies yourself, using the single-include header for catch is much easier.
- Singularity version of Python affects git hooks, but if you don't have the same version as the container it creates problems. ~Thus force python2 in the suggested build commands.~ So add a note explaining this so the user is aware.

## Proposed changes

<!--
At a high level, describe what this PR does.
-->

### Upgrade instructions

<!--
If this PR makes changes that other people should be aware of when upgrading
their code, describe what they should do between the two UPGRADE INSTRUCTIONS
lines below.
-->
<!-- UPGRADE INSTRUCTIONS -->

<!-- UPGRADE INSTRUCTIONS -->

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

knelli2 committed:
Update installation docs

Small things I found while setting up a new environment

- Blaze version was out of date.
- If installing catch from source, using the single-include
  header for catch is much easier.
- Update catch link and add note about not using v3
- Singularity version of Python affects git hooks, but if you don't
  have the same version as the container it creates problems. Add
  a note so that users are aware of this.
- Doxygen link was broken.

wthrowe commented on docs/Installation/Installation.md:
```diff
@@ -258,6 +260,10 @@ To build SpECTRE with Singularity you must:
 - You should edit source files in SPECTRE_ROOT in a separate terminal
   outside the container, and use the container only for compiling and running
   the code.
+- We force the use of python2 here because the Singularity container uses
+  python3.8 by default. However, unless you have python3.8 in your environment
+  outside the container, this will create problems with git hooks, thus it is
+  suggested to use cmake with python2 inside Singularity.
```
Some people won't have python 2 these days.  Distros are starting to drop it.

knelli2 replied:
So then do we just leave it up to the user to figure out their python dependencies?

nilsdeppe replied:
Pretty much. Most people should be using python3 nowadays, but it's annoying that python decides that any minor version change means they can change anything and everything in the language. Python stability basically disappeared in 3.x

knelli2 replied:
Ok instead of changing it in the example cmake command, I'll keep a note saying users should be aware of this.

nilsdeppe replied:
Yep, sounds reasonable to me :)

knelli2 commented:
Ok updated!

nilsdeppe reviewed (dismissed):
I'm happy with this. Thanks for updating this info :)

nilsvu commented on docs/Installation/Installation.md:
```diff
@@ -31,10 +31,10 @@ all of these dependencies.
 * [Boost](http://www.boost.org/) 1.60.0 or later
 * [Brigand](https://github.com/edouarda/brigand)
-* [Catch](https://github.com/philsquared/Catch) 2.8.0 or later
+* [Catch](https://github.com/philsquared/Catch) 2.8.0 or later (it is recommended to use single-header installation)
```
- Would you mind updating the link to `https://github.com/catchorg/Catch2`?
- I'm not so sure we should add this recommendation. For instance, for me it was easiest to install catch through my package manager (macOS). On Minerva I installed it with Spack. I think we can't give a general recommendation here how to install things, but only on the installation options (like "disable mpi" for HDF5 and libsharp below). Which issue did you run into that prompted you to add this note? If you feel like adding a note to help others with that issue, perhaps phrase it more specific to that issue?
- Note that Catch is developing v3, which will no longer be header-only: https://github.com/catchorg/Catch2/blob/devel/docs/migrate-v2-to-v3.md#top

knelli2 replied:
- The current link redirects you to Catch2, but I'll still update it
- I was installing all the dependencies on my machine and inside catch there is `include/catch.cpp` and `single_include/catch2/catch.hpp`. I tried the first option but got a lot of errors when compiling about missing headers. The second option (single include) is what we are supposed to use and what is used inside the containers I believe.
- When (if) we switch to v3, then we can update docs. I don't think that should matter for this because it's v2.

yoonso0-0 replied:
Their `devel` branch is already in v3 and has a bit different source structure with v2.x. Should we add a phrase like '2.8.0 or later but not 3.x'?

knelli2 replied:
As long as we don't yet support v3, then yeah I think that's ok.

nilsvu replied:
ok the way you formulated this looks fine to me. If you want to include @yoonso0-0's suggestion to exclude v3 then squash that in as well. No need to wait for my re-approval if someone gets to review this before me.

yoonso0-0 commented:
I found doxygen link also broken and would be good to include in this PR (`https://www.doxygen.nl/index.html`).

knelli2 commented:
@nilsleiffischer reworded that help phrase and updated catch link
@yoonso0-0 updated doxygen link

Rebased on latest develop and squash changes directly.

knelli2 commented:
Included @yoonso0-0 's suggestion of excluding catch v3
